Fixed latex export for svg data in python 3

MSeal opened the pull request:
Fixes https://github.com/jupyter/nbconvert/issues/981

Verified that the rendered latex output matches python 2 output visually. This will probably fix other python exports outside of latex svg given the code path involved in the fix.

akhmerov commented on nbconvert/preprocessors/extractoutput.py:
```diff
@@ -80,27 +80,28 @@ def preprocess_cell(self, cell, resources, cell_index):
-                        not isinstance(data, text_type)
-                        or mime_type == 'application/json'
-                    ):
+                    if mime_type == 'application/json':
```
This is too strict and will break some data types; compare with even a partial listing here: https://jupyter.readthedocs.io/en/latest/reference/mimetype.html

Also this change invalidates the follow-up comment, please keep them in sync.

MSeal replied:
Fair point on the strictness.

akhmerov commented:
In the original implementation I've relied on the nbformat spec that guaranteed that json data is parsed directly and not serialized.

I don't actually understand why that bit requires changing to support svg: IIUC svg should be of text-type and shouldn't be treated as json by that code.

MSeal commented:
If you only add the svg test I added you'll see if fails for python 3 in the same way the original issue describes. The `data` object coming back from `out.data[mime_type]` is definitely `bytes` and adding `data = data.decode('utf-8')`, or even the original `data = json.dumps(data)`, generically to all byte data before any other processing seemed wrong.

I'll look into the test case I added and see if there's a less strict way to solve the issue.

akhmerov commented:
At the very least lists/dicts should be detected as definitely json

MSeal commented:
I looked into the failing pattern, it's when the mimetype is `application/pdf` that it fails in the test case originally.

I changed the implementation to mirror the original behavior more closely, with just logically moving the base64 data type checks in front of the json / text checks. Can you take a look again @akhmerov for any issue you can see?

MSeal commented:
Thanks for the review!
